src/coreclr/src/debug/ee/debuggermodule.cpp

```diff
@@ -239,12 +239,34 @@ void DebuggerModuleTable::RemoveModule(Module* module, AppDomain *pAppDomain)
     }
     CONTRACTL_END;
 
-    // If this is a domain neutral module, then scan the complete list of DebuggerModules looking
-    // for the one with a matching appdomain id.
-    // Note: we have to make sure to lookup the module with the app domain parameter if the module lives in a shared
-    // assembly or the system assembly. <BUGNUM>Bugs 65943 & 81728.</BUGNUM>
-    _ASSERTE( FALSE );
+    LOG((LF_CORDB, LL_INFO1000, "DMT::RM Attempting to remove Module:0x%x AD:0x%x\n", pModule, pAppDomain));
 
+    _ASSERTE(ThreadHoldsLock());
+    _ASSERTE(pModule != NULL);
+
+    HASHFIND hf;
+    DebuggerModuleEntry *pDME; pDME = (DebuggerModuleEntry *) FindFirstEntry(&hf);
+
+    for (pDME = (DebuggerModuleEntry *) FindFirstEntry(&hf);
```

Comment:
Line 248 and line 250 should be simplified.  

Delete line 248 and...
```suggestion
    for (DebuggerModuleEntry pDME = (DebuggerModuleEntry *) FindFirstEntry(&hf);
```
or
```suggestion
    DebuggerModuleEntry pDME = (DebuggerModuleEntry *) FindFirstEntry(&hf);
    for (;
```
or 
```suggestion
    for (auto pDME = (DebuggerModuleEntry *) FindFirstEntry(&hf);
```

```

Reply:
Wow, I didn't even see this when I restructured the code. Thanks